### ISSUE TITLE
Terminate worker processes cleanly if main process terminates

### ISSUE THREAD

**Issue by prajnoha** [sid-project/sid#82]
Normally, when main process terminates, it also terminates its workers. But if there's any other unexpected exit of the main process, we need the workers to know - use `prctl(PR_SET_PDEATHSIG...)` to define a signal to which workers can react appropriately.

**Comment by trgill** [comment]
@prajnoha I looked at this and opened PR #99

Currently sidctl displays the following when SID is killed and the worked it is connected to terminates.

```
# sidctl dump --format json
<sidctl> Failed to read response: Connection reset by peer.
```

**Comment by prajnoha** [comment]
(Ah, sorry, I meant to write the comments here, but I wrote them in #99 instead - so please read there :) )

**Comment by prajnoha** [comment]
Fixed with #99.
